src/useCache.ts

```diff
@@ -1,17 +1,33 @@
-const cache: { [k: string]: { value: any; expiry: number } } = {}
-
 export const useCache = async (
+  KV: KVNamespace,
+  context: ExecutionContext,
   key: string,
   callback: Function,
   expirySeconds = 3600
 ) => {
-  const currentTime = new Date().valueOf() / 1000
-  const cached = cache[key]
-  if (cached && cached.expiry >= currentTime) return cached.value
-  cache[key] = { value: await callback(), expiry: currentTime + expirySeconds }
-  return cache[key].value
+  try {
+    const cached = await KV.get(key)
+    if (cached) return cached
+
+    const valueToCache = await callback()
+
+    if (valueToCache) {
+      context.waitUntil(
+        KV.put(key, valueToCache, {
```

Comment:
let's also make sure to `await` this call, for the same reason we're awaiting `set`